test/js-api/table/assertions.js

```diff
@@ -1,24 +1,29 @@
-function assert_equal_to_array(table, expected, message) {
-  assert_equals(table.length, expected.length, `${message}: length`);
+function assert_equal_to_array(table, expected, message, index = "i32") {
+  function idx(i) {
+    return index === "i64" ? BigInt(i) : i;
```

Comment:
This is the kind of thing we have to do all over the place in emscripten, its pretty annoying.